tests/sock_util/main.c

```diff
@@ -0,0 +1,75 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "sock_util.h"
+#include "warnp.h"
+
+static struct testcase_port {
+	const char * input;
+	const char * output;
+} tests_port[] = {
+	{ "/tmp/mysock", "/tmp/mysock" },
+	{ "localhost", "localhost:0" },
+	{ "localhost:80", "localhost:80" },
+	{ "[127.0.0.1]", "[127.0.0.1]:0" },
+	{ "[127.0.0.1]:80", "[127.0.0.1]:80" },
+	{ "[::1]", "[::1]:0" },
+	{ "[::1]:80", "[::1]:80" },
+	/* The below lines are not officially supported. */
+	{ "127.0.0.1", "127.0.0.1:0" },
```

Comment:
Since we're not officially supporting this form of address (as noted by the comment above), you might prefer it if I deleted them.  I think it's worth keeping, though, since many (most?) platforms will resolve "127.0.0.1" perfectly happily, and if spiped stopped accepting it then some people would report it as a bug.

The ipv6 form of "::1" (without `[]`) is not handled by this function, though.

Reply:
Let's take it out.  It only works because most DNS resolvers will resolve "127.0.0.1" to 127.0.0.1, but I don't think that's required and it could break if your DNS resolver is nonlocal anyway.